tensorboard/http_context.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Context to HTTP request properties."""
+
+
+class HttpContext:
+    """Container of values related to caller's HTTP request.
+
+    This context is for providing security, and quota concerns.
+
+    `HttpContext` values are to be treated as immutable.
+
+    Fields:
+      remote_ip: Best possible guess of the caller's IP Address.
+      x_forwarded_for: Value of X-Forwarded-For header.
+    """
+
+    def __init__(self, remote_ip=None, x_forwarded_for=None):
+        """Create a HTTP context.
+
+        The argument list is sorted and may be extended in the future;
+        therefore, callers must pass only named arguments to this
+        initializer.
+
+        Args: See "Fields" on class docstring. All arguments are optional and will
+        be replaced with default values if appropriate.
+        """
+        self._remote_ip = remote_ip
+        self._x_forwarded_for = x_forwarded_for
```

Comment:
Why introduce a new nested container instead of putting these on the
top-level context? It seems reasonable to me to put properties of the
HTTP request on the `RequestContext`, and it would be nice to avoid a
proliferation of types and modules if not necessary. It’s clear from the
docs and implementation that `HttpContext` and `RequestContext` have the
same structure, so it seems like an easy improvement to me.